tools/wpt/revlist.py

```diff
@@ -0,0 +1,299 @@
+        "type": bool
+    }
+}
+
+
+def get_git_cmd(repo_path):
+    # type: (bytes) -> Callable[..., Text]
+    """Create a function for invoking git commands as a subprocess."""
+    def git(cmd, *args):
+        # type: (Text, *Union[bytes, Text]) -> Text
+        arg_list = list(item.decode("utf8") if isinstance(item, bytes) else item for item in args)  # noqa E501
+        full_cmd = [u"git", cmd] + arg_list  # type: List[Text]
+        try:
+            logger.debug(" ".join(full_cmd))
+            return subprocess.check_output(full_cmd, cwd=repo_path).decode("utf8").strip()  # noqa E501
+        except subprocess.CalledProcessError as e:
+            logger.critical("Git command exited with status %i" % e.returncode)
+            logger.critical(e.output)
+            sys.exit(1)
+    return git
+
+
+def get_tagged_revisions(commit, min_age, skip=0, max_count=10):
+    # type: (bytes, int, int, int) -> Dict[..., Dict]
+    '''
+    Returns the tagged revisions indexed by the committer date.
+    '''
+    git = get_git_cmd(wpt_root)
+    command = [
+        "--no-walk",
+        "--skip=%s" % skip,
+        "--max-count=%s" % max_count,
+        "--min-age=%s" % min_age,
+        "--format=epoch:%ct:commit:%H:%D",
+        commit
+    ]
+    result = {}
+    for output_line in git("rev-list", *command).split("\n")[1::2]:
+        # print output_line
+        output = output_line.split(":")
+        if len(output) > 5 and output[5].find(TAGGED_REVISIONS) > -1:
+            revision = {}
+            revision["epoch"] = int(output[1])
+            revision["commit"] = output[3].strip()
+            revision["tag"] = output[5].strip()
+            result[int(output[1])] = revision
+    return result
+
+
+def get_newer_tagged_revision(tagged_revisions, min_age):
+    # type: (Dict, int) -> Dict
+    epoch = max(i for i in tagged_revisions.keys() if i < min_age)
+    c = tagged_revisions.keys()
+    c.sort(reverse=True)
+    return tagged_revisions[epoch]
+
+
+def list_tagged_revisons(**kwargs):
+    # type: (**Any) -> List[Text]
+    head_revision = COMMAND_ARGS["head_revision"]["parser"](**kwargs)
+    logger.debug("list_tagged_revisons: %s" % kwargs)
+    epoch_step = COMMAND_ARGS["epoch_step"]["parser"](**kwargs)
+    epoch_threshold = COMMAND_ARGS["epoch_threshold"]["parser"](**kwargs)
+    epoch_until = COMMAND_ARGS["epoch_until"]["parser"](**kwargs)
+    num_revisions = COMMAND_ARGS["num_revisions"]["parser"](**kwargs)
+    verbose = COMMAND_ARGS["verbose"]["parser"](**kwargs)
+
+    if epoch_until > DEFAULT_EPOCH_UNTIL - epoch_threshold:
+        epoch_until = DEFAULT_EPOCH_UNTIL - epoch_threshold
+        logger.debug("epoch-until adjusted to %s using the safety epoch-threshold value (%s)" % (epoch_until, epoch_threshold))  # noqa E501
+    iterator = 0
+    result = []
+    previous_revision = None
+
+    # Iterates the rev-list command in bunchs of "rev_list_step"
```

Comment:
Typo "bunchs" should be... bunches? But there's now just a single invocation of `git()` so this documentation isn't correct. Describing what the algorithm does at the level of the ASCII diagram would be helpful, even if there were chunking of the work I don't think it'd be necessary to say much about it here.